src/lib/components/event/event-history-timeline.svelte

```diff
@@ -112,88 +133,139 @@
-    timeline.setItems(items);
-    timeline.fit();
+    const reverseHistory =
+      $eventFilterSort === 'descending' && $eventViewType !== 'compact';
+    const sortedHistory = reverseHistory
+      ? [...history].reverse()
+      : [...history];
+    const filteredHistory = filterHistory(sortedHistory, category);
+    const eventGroups = groupEvents(filteredHistory);
+    const { groups, items } = createGroupItems(
+      eventGroups,
+      $workflowRun?.workflow?.isRunning,
+      sortedHistory,
+    );
+    setVizItems(items, groups);
   };
 
+  onMount(() => {
+    return () => timeline?.destroy();
+  });
+
+  $: readyToDraw =
+    $workflowTimelineViewOpen &&
+    $workflowRun.workflow &&
+    history.length &&
+    visualizationRef;
   $: {
-    if (history.length && timeline) {
-      const reverseHistory =
-        $eventFilterSort === 'descending' && $eventViewType !== 'compact';
-      const sortedHistory = reverseHistory
-        ? [...history].reverse()
-        : [...history];
-      const eventGroups = groupEvents(sortedHistory);
-      const { groups, items } = createGroupItems(
-        eventGroups,
-        $workflowRun?.workflow?.isRunning,
-        sortedHistory,
-      );
-      setVizItems(items, groups);
+    if (readyToDraw) {
+      if (timeline) {
+        timeline.destroy();
+      }
+      buildTimeline(category);
     }
   }
-  const resetTimelineView = () => {
-    timeline.focus('workflow');
-  };
 </script>
 
-<div
-  class="flex w-full flex-col gap-4 rounded-xl border-2 border-gray-900 bg-white p-4"
+<Accordion
+  title={translate('timeline')}
+  icon="timeline"
+  open={$workflowTimelineViewOpen}
+  onToggle={() => {
+    $workflowTimelineViewOpen = !$workflowTimelineViewOpen;
+  }}
 >
-  <div class="flex items-center justify-between gap-2">
-    <h3 class="text-xl">Timeline</h3>
-    <div class="flex gap-2">
-      <Button variant="secondary" on:click={() => timeline.zoomIn(1)}
-        >Zoom In</Button
-      >
-      <Button variant="secondary" on:click={() => timeline.zoomOut(1)}
-        >Zoom Out</Button
-      >
-      <Button variant="secondary" on:click={resetTimelineView}
-        >Zoom to Fit</Button
-      >
+  <div class="flex flex-col gap-2">
+    <div class="flex items-center justify-end gap-2">
+      <div class="flex gap-2">
+        <ToggleButtons>
+          <ToggleButton
+            data-testid="zoom-in"
+            on:click={() => timeline.zoomIn(1)}>+</ToggleButton
+          >
+          <ToggleButton
+            data-testid="zoom-in"
+            on:click={() => timeline.zoomOut(1)}>-</ToggleButton
+          >
+          <ToggleButton
+            data-testid="zoom-in"
+            on:click={() => timeline.focus('workflow')}>Fit</ToggleButton
+          >
+        </ToggleButtons>
+      </div>
     </div>
+    <div class="timeline" bind:this={visualizationRef} />
   </div>
-  <div bind:this={visualizationRef} />
-</div>
+</Accordion>
 
 <style lang="postcss">
```

Comment:
Noticed that on some `Completed` workflows you can zoom out and see a red moving line that correlates with the current time, which is a bit confusing since the workflow is `Completed` and there should be no more events 🤔 

![Screenshot 2023-10-02 at 2 54 38 PM](https://github.com/temporalio/ui/assets/15069288/94e5cd77-6876-4a39-9da8-d6ce831faa41)

Reply:
So this happens because I've set the max time to be the end of the minute of the minute that the workflow execution ends. So you'd only see that red moving line for the <60 seconds after it completes. I do this so you can zoom out a little to see any events that happen right at the end of the workflow.

Reply:
Ah, got it. Wanted to make sure I wasn't missing something 😅